app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -105,15 +99,42 @@ const TemplateRender: React.FC<TemplateRenderProps> = ({ template, onChange }) =
                                 />
                             </svg>
                         </button>
-                        <p className="font-[600] text-sm">{template.title}</p>
+                        <p className="font-[600] text-sm">New empty project</p>
                     </div>
                 </div>
-            )}
-        </button>
+            </button>
+        ),
+        []
+    )
+    return (
+        <>
+            {CreateEmptyTemplate}
+            {templates.map(template => (
+                <button
+                    key={template.title}
+                    className="h-40 cursor-pointer"
+                    onClick={() => {
+                        onChange(template.templateName)
+                    }}
+                >
+                    <div
+                        style={{ backgroundImage: genBackgroundImage() }}
+                        className="flex flex-col justify-end h-full w-full rounded-2xl overflow-hidden text-white text-left bg-cover"
+                    >
+                        <div className="bg-black bg-opacity-30 px-4 pt-2 pb-4">
+                            <div className="text-sm font-bold pb-2">{template.title}</div>
+                            <div className="text-xs h-[2lh] text-ellipsis">
+                                {template.description}
+                            </div>
+                        </div>
+                    </div>
+                </button>
+            ))}
+        </>
     )
 }
 
-/** Render all templates. */
+/** The TemplatesRender's container. */
 interface TemplatesProps {
     onChange(name?: string): void
```

Comment:
actually, not sure this is a good name - to me `onchange` is usually when the value of one single input changes. in this case, there is no previous value so there's nothing for it to change from
maybe `createProject` (describing what the callback should do), or `onTemplateClick` (describing what event triggers the callback). i'm not sure if we have an existing convention for how to name these props

Reply:
I don't know too, I'm following what the people before me wrote.
The onChange also he wrote.

Reply:
i think it wouldn't hurt to change it especially since you're changing so much of `templates.tsx`, but i guess it's fine to leave it as it is?